Fable.Remoting.MsgPack/Write.fs

```diff
@@ -367,8 +379,12 @@ and private makeSerializerAux<'T> (ctx: TypeGenerationContext): Action<'T, Strea
     | Shape.Tuple (:? ShapeTuple<'T> as shape) ->
         let elementSerializers = shape.Elements |> Array.map makeMemberVisitor
         Action<_, _> (fun (tuple: 'T) out -> writeTuple tuple out elementSerializers) |> w
-    | _ ->
-        failwithf "Cannot serialize %s." typeof<'T>.Name
+    | _shape ->
+        let tableTy =  typeof<System.Data.DataTable>
+        if typeof<'t> = tableTy ||  typeof<'t>.IsInstanceOfType tableTy then
+          Action<_, _> (fun (table: System.Data.DataTable) out -> writeDataTable table out) |> w
+        else
+          failwithf "Cannot serialize %s." typeof<'T>.Name
```

Comment:
There's nothing inherently wrong with this, but I'd rather define a shape recognizer (active pattern) for data tables to keep the big pattern match clean. You can base it on this one:

https://github.com/Zaid-Ajaj/Fable.Remoting/blob/7e3c52003c7390e57a8ea6e6f1e4afac3f12d199/Fable.Remoting.Server/Proxy.fs#L46-L49